source/common/quic/envoy_quic_connection_debug_visitor_factory_interface.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#ifndef THIRD_PARTY_ENVOY_SRC_SOURCE_COMMON_QUIC_ENVOY_QUIC_CONNECTION_DEBUG_VISITOR_FACTORY_INTERFACE_H_
+#define THIRD_PARTY_ENVOY_SRC_SOURCE_COMMON_QUIC_ENVOY_QUIC_CONNECTION_DEBUG_VISITOR_FACTORY_INTERFACE_H_
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/optref.h"
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+#include "envoy/server/process_context.h"
+
+#include "quiche/quic/core/quic_connection.h"
+#include "quiche/quic/core/quic_session.h"
+
+namespace Envoy {
+namespace Quic {
+
+class EnvoyQuicConnectionDebugVisitorFactoryInterface : public Config::TypedFactory {
```

Comment:
This name has a hilarious number of components :) I particularly enjoy the "VisitorFactoryInterface" suffix which is three different design patterns in a row. Remind me of my favorite class in Chrome `HttpStreamFactoryImplJobController`. Anyway, this is fine and the name makes sense. I just found it amusing :)

Reply:
Yeah its a bit ridiculous 😄